darwin_fiftyone/darwin.py

```diff
@@ -622,142 +637,113 @@ def _upload_annotations(self, label_schema, samples, media_field, dataset, backe
-                        ), "Video samples must have frame-level labels"
-
-                    # Checks for videos
-                    if sample.metadata is None:
-                        if is_video:
-                            sample.metadata = fom.VideoMetadata.build_for(
-                                sample[media_field]
-                            )
-                        else:
-                            sample.metadata = fom.ImageMetadata.build_for(
-                                sample[media_field]
-                            )
-                    logging.info(f"Sample metadata: {sample.metadata}")
+                if is_video:
+                    assert (
+                        sample.frames
+                    ), "Video samples must have frames. Please use the ensure_frames() method on your dataset or view to ensure that frames are available."
+                    assert all(
+                        lf.startswith("frames") for lf in label_schema.keys()
+                    ), "Video samples must have frame-level labels (e.g. 'frames.detections')"
 
+                # Checks for videos
+                if sample.metadata is None:
                     if is_video:
-                        frame_size = (
-                            sample.metadata.frame_width,
-                            sample.metadata.frame_height,
-                        )
+                        sample.metadata = fom.VideoMetadata.build_for(sample[media_field])
                     else:
-                        frame_size = (sample.metadata.width, sample.metadata.height)
+                        sample.metadata = fom.ImageMetadata.build_for(sample[media_field])
+                logging.info(f"Sample metadata: {sample.metadata}")
 
-                    file_name = Path(sample[media_field]).name
-                    darwin_annotations = []
-
-                    if is_video:
-                        frames = {}
-                        frame_id_map[sample.id] = {}
+                if is_video:
+                    frame_size = (
+                        sample.metadata.frame_width,
+                        sample.metadata.frame_height,
+                    )
+                else:
+                    frame_size = (sample.metadata.width, sample.metadata.height)
 
-                        for frame_number, frame in sample.frames.items():
-                            frame_id_map[sample.id][str(frame_number)] = frame.id
+                file_name = Path(sample[media_field]).name
+                darwin_annotations = []
 
-                            if frame_number not in frames:
-                                frames[frame_number] = []
+                if is_video:
+                    frame_id_map[str(sample.id)] = {}
 
-                            frame_val = frame
+                    for frame_number, frame in sample.frames.items():
+                        frame_id_map[str(sample.id)][str(frame_number)] = str(frame.id)
```

Comment:
The stringification of `sample.id` looks a bit risky, as in other parts it is used as is. I'm afraid of potential mismatch in other code parts. Could you please double check it?